lib/elixir/lib/enum.ex

```diff
@@ -1436,6 +1436,20 @@ defmodule Enum do
   end
 
   @doc """
+  Invokes `fun` for each element in the collection passing that element and the
+  accumulator `acc` as arguments. `fun`'s return value is expected to be
+  `{:cont, acc}`, return `{:halt, acc} to end the reduction.
+  Returns the accumulator.
+
+  ## Examples
+    iex> Enum.reduce_while(1..100, 0, &(if &1 <= 3, do: {:cont, &2 + &1}, else: {:halt, &2}))
```

Comment:
I could not for the life of me get a mutli-line `fun` example to pass doctest, what's the trick?